projects/feature-activation/0001-feature-activation.md

```diff
@@ -0,0 +1,558 @@
+
+## Evaluation interval
+
+The evaluation interval, as explained in the [Evaluation interval] section, is defined here to be two weeks (14 days). Considering the 30 seconds average time between blocks of the Hathor network, that interval is equivalent to 40320 blocks. A constant must be defined in `HathorSettings`:
+
+```python
+EVALUATION_INTERVAL = 40320
+```
+
+## Criteria configuration
+[Criteria configuration]: #criteria-configuration
+
+The `Criteria` data class represents the configuration for a certain feature activation criteria:
+
+```python
+class Criteria(BaseModel):
+    name: str
+    bit: int
+    start_height: int
+    timeout_height: int
+    threshold: int
+    minimum_activation_height: int
+    activate_on_timeout: bool
+    version: str
+```
+
+### Attribute descriptions
+
+#### `name`
+
+A string representing the name of the feature. Must be unique across all features, past and future.
+
+#### `bit`
+[bit]: #bit
+
+Represents which bit in the `version` field of the block is going to be used to signal the feature support by miners. Must be chosen from the set {0,1,2,3}, as defined in the [Bit signaling implementation] section.
+
+Must be selected such that no two concurrent features use the same bit. Also, a bit should only be reused if the `start_height` of a new feature is greater than the `timeout_height` of the previous feature that used this bit.
+
+#### `start_height`
+
+Specifies the height of the first block at which this feature's activation process starts. Must be some block height in the future.
+
+Must be an exact multiple of `EVALUATION_INTERVAL` (that is, at an evaluation interval boundary).
+
+#### `timeout_height`
+
+Specifies the height of the first block at which this feature's activation process is over. Once this height has been reached, if the feature is not activated and `activate_on_timeout` is `False`, the activation is considered failed on all descendants of the block.
+
+Should be some height greater than `start_height`. Should be an exact multiple of `EVALUATION_INTERVAL` (that is, at an evaluation interval boundary).
+
+#### `threshold`
+
+Specifies the minimum number of blocks per evaluation interval required to activate the feature. Usually calculated from a percentage of `EVALUATION_INTERVAL`.
+
+Here it is defined as 36288 blocks (90% of `EVALUATION_INTERVAL`) for the mainnet and 30240 blocks (75% of `EVALUATION_INTERVAL`) for the testnet, as default values. It may be different for each feature, though.
+
+#### `minimum_activation_height`
+
+Specifies the height of the first block at which the feature is allowed to become active. If it's smaller than `start_height` (usually 0), then there's no restriction on when this feature can become active.
+
+Must be less than or equal to `timeout_height`. Must be an exact multiple of `EVALUATION_INTERVAL` (that is, at an evaluation interval boundary).
```

Comment:
Should we create a `validate()` method for the feature flags? It seems we can perform some verifications in the CI or as part of the release process.

Reply:
Makes sense. Internal rules will be validated by the `Criteria` class itself, for example, checking that `minimum_activation_height` is less than or equal to `timeout_height` and a multiple of `EVALUATION_INTERVAL`.

That `validate()` method would be useful to check for rules between different features, like checking that no two features are using the same name, or the same bit concurrently. I'll add it to the design.

Reply:
Done in https://github.com/HathorNetwork/rfcs/pull/51/commits/1230bbad8164aaf441b9ba70009b84f734f7d973

Reply:
Great! Thanks!